pkg/store/event/event.go

```diff
@@ -341,8 +375,15 @@ func (e *eventStore) DryListByFilter(_ context.Context, filter core.Filter) ([]*
 		resList []*core.Event
 		err     error
 		db      *dbstore.DB
+		limit   int
 	)
 
+	if filter.LimitStr != "" {
```

Comment:
The two functions `DryListByFilter` and `ListByFilter` has many duplicates because they share the same logic, can we abstract some of them to another function?

Reply:
"DryListByFilter" is mainly used by grafana, and may be accessed more frequently than "ListByFilter". In order not to affect "ListByFilter", it is split out. To merge into the same function is not complicated to implement, mainly based on stability considerations.

Reply:
I am not saying to combine the two functions into one. But the two share some parts of duplicate code (except the pod records part). Is it okay to pull the duplicate part out and make it another function?

Reply:
But this actually is not part of this pr, so we can have an issue for it first and not do it in this pr. It depends on you

Reply:
Oh oh, I got it. I just got it wrong. Let me change it in the follow-up pr.